contracts/ERC721A.sol

```diff
@@ -141,20 +148,37 @@ contract ERC721A is IERC721A {
         _name = name_;
         _symbol = symbol_;
         _currentIndex = _startTokenId();
+
+        if (_sequentialUpTo() < _startTokenId()) _revert(SequentialUpToTooSmall.selector);
     }
 
     // =============================================================
     //                   TOKEN COUNTING OPERATIONS
     // =============================================================
 
     /**
-     * @dev Returns the starting token ID.
-     * To change the starting token ID, please override this function.
+     * @dev Returns the starting token ID for sequential mints.
+     *
+     * Override this function to change the starting token ID for sequential mints.
+     *
+     * Note: The value returned must never change after any tokens have been minted.
      */
     function _startTokenId() internal view virtual returns (uint256) {
         return 0;
     }
 
+    /**
+     * @dev Returns the maximum token ID (inclusive) for sequential mints.
+     *
+     * Override this function to return a value less than 2**256 - 1,
+     * but greater than `_startTokenId()`, to enable spot (non-sequential) mints.
+     *
+     * Note: The value returned must never change after any tokens have been minted.
+     */
+    function _sequentialUpTo() internal view virtual returns (uint256) {
```

Comment:
Thoughts on making this a pure function? It would prevent the value from being dynamic, which would prevent edge cases.

Reply:
Pure functions can't read immutables and will make testing harder. 

Imo, it's ok to keep it view. `_startTokenId()` has been view all along, and has been used safely.